### ISSUE TITLE
There should be some 'blocking' and 'non-blocking' errors in the CLI `lightdash preview`

### ISSUE THREAD

**Issue by TuringLovesDeathMetal** [lightdash/lightdash#2552]
### Description

Right now, if I'm developing my Lightdash project, I might not have all of the tables in my project existing in my development space. I'll probably just have the tables I'm working on that are in BigQuery (e.g. I'm working on `users`, I'll probably have everything upstream + downstream from that in my available tables)

But, the Lightdash CLI won't let me preview the instance if I don't have _**all**_ of the tables in my project available. 
<img width="1194" alt="image" src="https://user-images.githubusercontent.com/31848148/175552289-edd66645-3760-461b-b8a4-db1532063c41.png">

Basically, I'd expect the preview to work the same way as my current project: if some tables are missing, they'd just be greyed out (and I might get some notification of this on the CLI tool). If the project is broken, I would get an error saying that the project was broken and I couldn't preview it. 

 ### What problem does this solve?

I can't really use `lightdash preview` when I'm developing if I'm working in a huge project.

**Comment by owlas** [comment]
@TuringLovesDeathMetal good catch! 

On this one, how would we know whether a model should be happily greyed out because the user doesn't care about it or because the user has made a configuration error (maybe the schema name etc.) and actually wants to catch that at the CLI before pushing to Lightdash?

**Comment by TuringLovesDeathMetal** [comment]
I'd say that we should tell them the errors, but still let them click it! 

Does that makes sense? 

E.g. I've been changing `organizations` and made a typo. 

I hit `lightdash preview` and it says `⚠️ warning - there are these errors in your project.` But I'm still able to click and open it

**Comment by owlas** [comment]
If you were doing `lightdash deploy` to prod, would you still want this behaviour? Or would you prefer it to stop deployment after hitting any error?

**Comment by TuringLovesDeathMetal** [comment]
I think I'd still want this behaviour (this may be quite controversial...)

But basically, unless you're a massive company, breaking BI isn't life or death. You want to be able to be fast and have the ability to "break stuff, and be aware that it's broken, but ship it anyways". 

Ideal scenario would be to have a settings option where you could turn off/on this for `lightdash deploy` (i.e. you could have it deploy with errors _**or**_ require that there are no errors for it to deploy successfully)

**Comment by owlas** [comment]
> I think I'd still want this behaviour (this may be quite controversial...)

👀 I'll say!!!

---

In this case the error comes from automatically figuring out the dimension and there's an inconsistency because when users hit "refresh dbt" in the UI we assume that missing dimensions are of string type. But it seems like there's a bug in the CLI where it's still returning errors instead of just defaulting to strings.

I think that's the most common error we've seen among `lightdash preview` users too

**Comment by TuringLovesDeathMetal** [comment]
cool - makes sense. 

So should I create a different ticket for "default to string type" dimensions? 

I still feel like being able to play around with things even if they're broken is something that's quite important in the `preview`. WDYT?

**Comment by owlas** [comment]
Yeah I think it depends. Whatever we need to speed up dev I'm for it. 

I can add a new ticket. Also I think this is very closely related to #2499

**Comment by TuringLovesDeathMetal** [comment]
@owlas I'm still running into this problem when developing in the CLI. I'm not sure if I've done something that wasn't changed in this fix 🤔 

In my example, I've been updating a model called `tracks` (not at all related to `users_daily`). Here's what I've done:

-  I've run everything upstream + including `tracks` in my dbt dev environment. So, everything I want to check out in Lightdash exists as a table in `dbt_khindson` in BigQuery. 
- My CLI tool is connected to my `Katie Dev` project in Lightdash (which is connected to `dbt_khindson` schema in BigQuery...not sure if this bit is essential 🤔 )
- I run `lightdash preview`
- It gives me an error:
```shell
Column "is_reactivated_on_date" from model "users_daily" does not exist.
 "users_daily.is_reactivated_on_date" was not found in your target warehouse at lightdash-analytics.dbt_khindson.users_daily. Try rerunning dbt to update your warehouse.
```
- I can't open the preview...which is annoying, because I didn't touch `users_daily` in my development. 

The ideal scenario:

I'd want this error to pop up, but then below it, I'd want it to still give me my preview (even if there were errors). 

I'd expect when I opened the preview, the tables with errors would be greyed out like they are in a real Lightdash instance.

**Comment by TuringLovesDeathMetal** [comment]
Reopening this!

I'm still running into problems when I'm trying to run `lightdash preview` where it's not letting me preview because there are unrelated models that are broken (i.e. models I'm not developing at the moment)

<img width="1272" alt="image" src="https://user-images.githubusercontent.com/31848148/180404631-7ed064d2-c509-483a-97e6-11e30b838924.png">


I explained the ideal scenario above, but I'm not sure if it's possible! This is one of the biggest blockers from me using `lightdash preview` regularly when I'm developing

**Comment by rephus** [comment]
I'm not sure exactly how to replicate an error like that. I added a new column on the schema that doesn't exist on the warehouse. DBT errors but preview works

![Screenshot from 2022-09-15 09-55-03](https://user-images.githubusercontent.com/1983672/190348135-feec7e0c-a584-4d3a-946e-8d91bea03c6b.png)
![Screenshot from 2022-09-15 09-54-28](https://user-images.githubusercontent.com/1983672/190348128-c89d81a3-41c9-4460-b3d8-acbef85f9136.png)

**Comment by TuringLovesDeathMetal** [comment]
oh nice! Okay, maybe this got fixed - I'm not sure...i'll see if I can replicate it again somewhere 👀

**Comment by rephus** [comment]
@TuringLovesDeathMetal I also tested this suggestion from Oliver:  deleting both orders and stg_orders on my local jaffle, I got an error on dbt run, preview works, I can't see the table

![image](https://user-images.githubusercontent.com/1983672/190371116-b9881f7b-2053-4ebd-84ec-237cce9a69a3.png)
![image](https://user-images.githubusercontent.com/1983672/190371140-c1bfc1cd-9e9d-4be9-9047-385784135fd4.png)
![image](https://user-images.githubusercontent.com/1983672/190371160-4e03e97c-7919-4ffc-a4e4-a0689df1ecd5.png)

**Comment by TuringLovesDeathMetal** [comment]
Nice! Ya, just tested it too - it's working now!
